src/index.ts

```diff
@@ -0,0 +1,213 @@
+ */
+class TranscriptionEditor {
+    anno;
+
+    annotationContainer: HTMLElement;
+
+    storage;
+
+    // TODO: Add typedefs for the Annotorious client (anno) and storage plugin
+
+    /**
+     * Initialize an instance of the TranscriptionEditor.
+     *
+     * @param {any} anno Instance of Annotorious.
+     * @param {any} storage Storage plugin to save Annotorious annotations.
+     * @param {HTMLElement} annotationContainer Existing HTML element that the editor will be
+     * placed into.
+     */
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    constructor(anno: any, storage: any, annotationContainer: HTMLElement) {
+        this.anno = anno;
+        this.storage = storage;
+        // disable the default annotorious editor (headless mode)
+        this.anno.disableEditor = true;
+        this.annotationContainer = annotationContainer;
+
+        // define custom elements
+        customElements.define("save-button", SaveButton, { extends: "button" });
+        customElements.define("cancel-button", CancelButton, {
+            extends: "button",
+        });
+        customElements.define("delete-button", DeleteButton, {
+            extends: "button",
+        });
+        customElements.define("annotation-block", AnnotationBlock, {
+            extends: "div",
+        });
+
+        // attach event listeners
+        document.addEventListener(
+            "annotations-loaded",
+            this.handleAnnotationsLoaded.bind(this),
+        );
+        this.anno.on("createSelection", this.handleCreateSelection.bind(this));
+        this.anno.on(
+            "selectAnnotation",
+            this.handleSelectAnnotation.bind(this),
+        );
+    }
+
+    /**
+     * Handler for custom annotations loaded event triggered by storage plugin.
+     */
+    handleAnnotationsLoaded() {
+        // remove any existing annotation displays, in case of update
+        this.annotationContainer
+            .querySelectorAll(".annotation-display-container")
+            .forEach((el) => el.remove());
+        // display all current annotations
+        this.anno.getAnnotations().forEach((annotation: Annotation) => {
+            this.annotationContainer.append(
+                new AnnotationBlock({
+                    annotation,
+                    editable: false,
+                    onCancel: this.anno.cancelSelected,
+                    onClick: this.handleClickAnnotationBlock.bind(this),
+                    onDelete: this.handleDeleteAnnotation.bind(this),
+                    onSave: this.handleSaveAnnotation.bind(this),
+                    updateAnnotorious: this.anno.addAnnotation,
+                }),
+            );
+        });
+    }
+
+    /**
+     * Instantiates an editable annotation block when a new selection is made.
+     *
+     * @param {Annotation} selection Selected Annotorious annotation.
+     */
+    async handleCreateSelection(selection: Annotation) {
```

Comment:
technically this returns a selection, not a full annotation https://recogito.github.io/annotorious/api-docs/osd-plugin/#createselection

I'm not sure how much the difference matters for us, and creating new annotations does seem to be working, but I noticed this when I was looking into the problem you identified.

Reply:
I noticed that nomenclature too… I don't know the whole W3C Web Annotation spec, but I wasn't able to find anything with `"type": "Selection"` as Annotorious sets it. And other than the `type` field it seems to conform to the spec of an Annotation ([example](https://www.w3.org/TR/annotation-model/#sets-of-bodies-and-targets)). (I did find [this](https://www.w3.org/TR/selection-api/#selection-interface) but it seems unrelated.)